test-data/unit/check-enum.test

```diff
@@ -1457,6 +1457,17 @@ class Foo(Enum):
 x = 4
 [builtins fixtures/bool.pyi]
 
+[case testEnumValueWithPlaceholderNodeType]
+# https://github.com/python/mypy/issues/11971
+from enum import Enum
+from typing import Callable, Dict
+
+class Foo(Enum):
+    Bar: Foo = Callable[[str], None]  # E: Value of type "int" is not indexable
```

Comment:
I don't quite understand this error which mypy is raising regarding `int` not being indexable. How is this relevant here?
Or is this due to some default of `Enum` assuming its values are of type `int`? :thinking:

Reply:
For completeness, this is the output of this test failing on `master` @ 6c1eb5b36b83ecf959a50100d282ca86186f470f:
```
E   AssertionError: Unexpected type checker output (/home/max/Git/mypy/test-data/unit/check-enum.test, line 1460)
------------------------------------------------- Captured stderr call -------------------------------------------------
Expected:
  main:6: error: Value of type "int" is not indexable
  main:7: error: Value of type "int" is not indexable (diff)
  main:7: error: Name "Missing" is not defined  (diff)
Actual:
  main:6: error: Value of type "int" is not indexable
  main:7: error: Attempted to reuse member name "Baz" in Enum definition "Foo" (diff)
  main:7: error: Value of type "int" is not indexable (diff)
  main:7: error: Name "Missing" is not defined  (diff)

Alignment of first line difference:
  E: main:7: error: Value of type "int" is not indexable...
  A: main:7: error: Attempted to reuse member name "Baz" in Enum definition "...
                    ^
```

Reply:
`Callable` is defined as `Callble = 0` in our test fixtures. You can define your own type instead for tests:

```python
class Some(Generic[T]): ...
``` 

It should not really matter.

Reply:
Alright then I assume it is fine as is once the CI passes. If not, please let me know